### DISCUSSION TITLE
iPhones with multiple camera array /small QR detection

### DISCUSSION THREAD

**Discussion by nbennettio** [juliansteenbakker/mobile_scanner#615] [Q&A]
with recent iPhone models (I have a iphone 11) the code uses the wide-angle camera instead of the main camera. This reduces the QR recognition capability. As a result, it becam almost impssible for me to achieve a good reading with smaller QR codes, which i use. 
With qr_code_scanner 1.0.1, which i was using previously, I had no problems reading small QR codes.
Anybody with an idea? Is it possible to choose which rear camera to use for the QR detection?

**Comment by ptrckdev** [comment]
The code actually lets Apple decide which camera to use. If you try to focus something at a distance it should switch to the main shooter.

Comically on even more recent iPhone models you _need_ to use the wide-angle camera as the minimum focus distance of the main shooter on the iPhone 14 Pro is too long (20cm) to focus on small QR Codes. There the automatic switching is needed to scan my 2cm wide QR Codes as that was not possible with the qr_code_scanner library I used before.
